<launch>

<group ns="bot1">

  <node pkg="turtlesim" type="turtle_teleop_key" name="teleop" output="screen">
  	<remap from="/bot1/turtle1/cmd_vel" to="/bot1/cmd_vel"/>
  </node>

  <node pkg="tftest" type="bot_node" name="body" output="screen">
	<param name="bot_name" value="bot1"/>
  </node>

  <node pkg="tftest" type="link_node" name="link1" output="screen">
  	<param name="link_name" value="link1"/>
        <param name="prev_link_name" value="body"/>
	<param name="bot_name" value="bot1"/>
  </node>

  <node pkg="tftest" type="link_node" name="link2" output="screen">
  	<param name="link_name" value="link2"/>
        <param name="prev_link_name" value="link1"/>
	<param name="bot_name" value="bot1"/>
  </node>

  <node pkg="tftest" type="link_node" name="link3" output="screen">
  	<param name="link_name" value="link3"/>
        <param name="prev_link_name" value="link2"/>
	<param name="bot_name" value="bot1"/>
  </node>


</group>


<group ns="bot2">

  <node pkg="tftest" type="ai_node" name="body" output="screen">
        <param name="bot_name" value="bot2"/>
        <param name="enemy_name" value="bot1"/>
  </node>

  <node pkg="tftest" type="link_node" name="link1" output="screen">
        <param name="link_name" value="link1"/>
        <param name="prev_link_name" value="body"/>
        <param name="bot_name" value="bot2"/>
  </node>

  <node pkg="tftest" type="link_node" name="link2" output="screen">
        <param name="link_name" value="link2"/>
        <param name="prev_link_name" value="link1"/>
        <param name="bot_name" value="bot2"/>
  </node>

  <node pkg="tftest" type="link_node" name="link3" output="screen">
        <param name="link_name" value="link3"/>
        <param name="prev_link_name" value="link2"/>
        <param name="bot_name" value="bot2"/>
  </node>


</group>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find tftest)/include/rviz_config.rviz"/>

</launch>
